dask/array/ghost.py

```diff
@@ -296,9 +344,48 @@ def ghost(x, depth, boundary):
            [100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100],
            [100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100]])
     """
+    if isinstance(depth, int):
+        depth = (depth,) * x.ndim
+    if isinstance(depth, tuple):
+        depth = dict(zip(range(x.ndim), depth))
+
+    if boundary is None:
+        boundary = 'reflect'
+    if not isinstance(boundary, (tuple, dict)):
+        boundary = (boundary,) * x.ndim
+    if isinstance(boundary, tuple):
+        boundary = dict(zip(range(x.ndim), boundary))
+
+    # is depth larger than chunk size?
+    for d, c in zip(depth.values(), x.chunks):
+        if d > min(c):
+            raise ValueError("Depth %d larger than chunk size %d"
+                             " to avoid this try to `rechunk` your array" %
+                             (d, min(c)))
```

Comment:
Suggest the following text:

    The overlapping depth %d is larger than your smallest chunk size %d.  
    Rechunk your array with a larger chunk size or a chunk size that more
    evenly divides the shape of your array.